CONVENTIONS.md

```diff
@@ -29,6 +29,53 @@ Black is left at defaults, flake8 and isort are configured to adhere to black.
 
 All functions need to be covered by unit tests.
 
+# Coordinate frame conventions
+
+As the purpose of pylinalg as a library primarily is to support linear algebra
+operations in pygfx and applications based on pygfx, we adhere to a number of
+standard coordinate frame conventions to align with expectations from pygfx.
+
+Generally there are four transform steps when going from an object's local space,
+all the way to screen space:
+
+* Local space to world space: world matrix
+* World space to view space: view matrix
+* View space to NDC/clip space: projection matrix
+
+In pygfx, every world object has its own model matrix, which
+encapsulates scale, rotation and translation components. World objects
+are additionally organized hierarchically using parent/child relationships.
+The world matrix is then recursively defined for every world object; it is the
+result of multiplying an object's model matrix with its parent's world matrix.
+For the root world object, the world matrix is equal to its model matrix. 
+
+The view and projection matrices are owned by camera objects. The view
+matrix can be acquired by inverting the camera's world matrix. The projection matrix
+is configured independently on the camera and depends on its
+type (perspective, orthographic).
+
+Pylinalg provides methods in its functional API to produce all of these matrices.
+Since world matrices are of key interest to users and developers working with pygfx
+there need to be some conventions in place regarding its axes.
+
+* The positive Z axis is the forward direction.
+* The positive Y axis is the up direction.
+* The positive X axis is the right direction.
+
+This means that if, for example, you add a car object to a scene, by default
+it is assumed that the car is facing towards positive Z. Due to the above axis
+conventions, you can then use `matrix_make_look_at` to easily create a new rotation
+matrix that makes the car look in another direction.
+
+# Memory layout conventions
+
+Row-major can mean two things:
+
+* Memory layout; are rows or columns contiguous in memory
+* Are vectors columns or rows
+
+Numpy is row-major in both cases, and so we adhere to the same
+convention here in pylinalg.
```

Comment:
```suggestion
Pylinalg's kernels are written assuming a row-major (C-contigous) layout. If a
kernel supports batch processing of vetors, it assumes that the last dimension
contains the relevant vector data and that all other dimensions are batch
dimensions. As such, you can think of vectors being row vectors.
```

Actually, numpy doesn't have a notion of a vector being a row or column vector. 

The community just happens to stack data along the first axis by convention, which has its roots in good old SQL, where each row represents a data tuple. scikit-learn stuck to that convention, and now it's everywhere. Numpy terminology talks about `core` and `loop` dimensions for batched processing, but I don't think we need to introduce this notion.